pkg/events/event.go

```diff
@@ -0,0 +1,287 @@
+package events
+
+import (
+	"sync"
+	"sync/atomic"
+
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+//
+// Event
+//
+
+// Event is a struct describing an event configuration.
+type Event struct {
+	id           *atomic.Uint32
+	id32Bit      *atomic.Uint32
+	name         string
+	docPath      string
+	strMutex     *sync.RWMutex
```

Comment:
`strMutex` as in for name and docpath? Not sure these should be mutable at all.

Reply:
ditto (about now knowing what should or shouldn't be changed runtime). There are things that CLEARLY shouldn't: ksymbol name, it is literally the "handle" that describes a ksymbol dependency and won't change *at all*. But many of other fields are "debatable".